src/App.tsx

```diff
@@ -1,24 +1,223 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
 
-const USER_ID = 0;
+import { Todo } from './types/Todo';
+import * as todoService from './api/todos';
+import { ForComletedTodo } from './types/enumFilter';
+import { TodoItem } from './Components/TodoItem';
+import { Footer } from './Components/Footer';
+import { Header } from './Components/Header';
+import { ErrorNotification } from './Components/ErrorNotification';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [condition, setCondition] = useState(ForComletedTodo.All);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [inputDisabled, setInputDisabled] = useState(false);
+  const [processingTodoIds, setProcessingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const hasTodos = todos.length > 0;
+
+  const isAllCompleted = todos.every(todo => todo.completed);
+
+  const fetchData = async () => {
+    try {
+      setErrorMessage(null);
+      const todosFetch = await todoService.getTodos();
+
+      setTodos(todosFetch);
+    } catch (err) {
+      setErrorMessage('Unable to load todos');
+    }
+  };
+
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const hadleAddTodo = (title: string) => {
+    setInputDisabled(true);
+
+    setTempTodo({
+      id: 0,
+      userId: 0,
+      title: title.trim(),
+      completed: false,
+    });
+
+    return todoService
+      .addTodo(title)
+      .then((newTodo) => {
+        setTodos((prevTodos) => [...prevTodos, newTodo]);
+      })
+      .catch((error) => {
+        setErrorMessage('Unable to add a todo');
+        throw error;
+      })
+      .finally(() => {
+        setInputDisabled(false);
+        setTempTodo(null);
+      });
+  };
+
+  const filteredTodos = useMemo(() => todos.filter(({ completed }) => {
+    switch (condition) {
+      case ForComletedTodo.Active:
+        return !completed;
+      case ForComletedTodo.Completed:
+        return completed;
+      default:
+        return 1;
+    }
+  }), [condition, todos]);
+
+  const handleDeleteTodo = (todoId: number) => {
+    setInputDisabled(true);
+    setProcessingTodoIds((prevTodoIds) => [...prevTodoIds, todoId]);
+
+    return todoService
+      .deleteTodo(todoId)
+      .then((() => {
+        setTodos((prevTodos) => prevTodos.filter(todo => todo.id !== todoId));
+      }))
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        throw new Error();
```

Comment:
You should pass the error text to the constructor of the `Error` class, otherwise what is the point of using it.
Fix this in all code.